src/shims/tls.rs

```diff
@@ -202,6 +191,8 @@ impl<'tcx> TlsData<'tcx> {
         for TlsEntry { data, .. } in self.keys.values_mut() {
             data.remove(&thread_id);
         }
+
+        self.macos_thread_dtors.remove(&thread_id);
```

Comment:
Aren't these always empty because we ran the destructors?

Reply:
Yes, but it seems neater to free the memory of the `Vec` here.

Reply:
Can you add an assertion ensuring the vector we pop is empty?